iocore/net/SSLNetVConnection.cc

```diff
@@ -1812,6 +1813,36 @@ SSLNetVConnection::populate(Connection &con, Continuation *c, void *arg)
   return EVENT_DONE;
 }
 
+void
+SSLNetVConnection::increment_ssl_version_metric(const char *version) const
+{
+  if (version) {
+    // openSSL guarantees the case of the protocol string.
+    if (version[0] == 'T' && version[1] == 'L' && version[2] == 'S' && version[3] == 'v' && version[4] == '1') {
+      if (version[5] == 0) {
+        SSL_INCREMENT_DYN_STAT(ssl_total_tlsv1);
+      } else if (version[5] == '.' && version[7] == 0) {
+        switch (version[6]) {
+        case '1':
+          SSL_INCREMENT_DYN_STAT(ssl_total_tlsv11);
+          break;
+        case '2':
+          SSL_INCREMENT_DYN_STAT(ssl_total_tlsv12);
+          break;
+        case '3':
+          SSL_INCREMENT_DYN_STAT(ssl_total_tlsv13);
+          break;
+        default:
+          break;
+        }
+      }
+    }
+  } else if (version[0] == 'S' && version[1] == 'S' && version[2] == 'L' && version[3] == 'v' && version[4] == '3' &&
```

Comment:
ditto

Reply:
This `else if` statement should be for `if (version[0] == 'T' && version[1] == 'L' && version[2] == 'S' && version[3] == 'v' && version[4] == '1') {` but not `if (version) {`. Clang-Analyzer complains that `version` can be `nullptr`.

I hope it will be gone by replacing this with a `switch`.

Reply:
+1, Do you know which include brings in `TLS1_3_VERSION`? I'm getting a compile failure for that symbol